src/compiler/program.ts

```diff
@@ -904,7 +904,13 @@ namespace ts {
             let declarationDiagnostics: Diagnostic[] = [];
 
             if (options.noEmit) {
-                return { diagnostics: declarationDiagnostics, sourceMaps: undefined, emittedFiles: undefined, emitSkipped: true };
+                // Allow only emitting of declarations
+                if (options.declaration) {

```

Comment:
I'm not sure I understand what you mean? Care to give an example?

The main usecase for `only declarations` is you have a pure js node project and you don't want to maintain hand typed .d.ts files.  You don't care about emitting since your code is already in js.